services/datalad/datalad_service/tasks/snapshots.py

```diff
@@ -132,6 +133,16 @@ def validate_snapshot_name(store, dataset, snapshot):
             'Tag "{}" already exists, name conflict'.format(snapshot))
 
 
+def validate_datalad_config(store, dataset):
+    """Add a .datalad/config file if one does not exist."""
+    dataset_path = store.get_dataset_path(dataset)
+    try:
+        git_show(dataset_path, 'HEAD', '.datalad/config')
+    except:
```

Comment:
Do we know what kind of exception to expect here? Bare `except`s are generally considered bad practice.

Reply:
Testing it, I can narrow it to pygit2.GitError. The other exceptions here should raise instead of just overwriting the file (invalid decoding or missing git objects).

Reply:
Actually KeyError is better, GitError just happens when there's some other issue (read error or a locked repo) rather than the file not existing in the commit referenced, and we probably don't want to try to replace it in that case either.